spectacles/lookml.py

```diff
@@ -322,14 +325,46 @@ def get_explore(self, model: str, name: str) -> Optional[Explore]:
             return model_object.get_explore(name)
 
     def get_results(
-        self, validator: str, fail_fast: Optional[bool] = None
+        self,
+        validator: str,
+        fail_fast: Optional[bool] = None,
+        filters: Optional[List[str]] = None,
     ) -> Dict[str, Any]:
         errors: List[Dict[str, Any]] = []
         successes: List[Dict[str, Any]] = []
         tested = []
 
         for model in self.models:
+            # Add model level content validation errors.
+            # We create an explore "tested" record for those that
+            # aren't in the LookML tree.
+            distinct_explores = set()
+            if filters is None and model.errors:
+                # Raise name error if there are errors on the model and filters
+                # isn't being passed. We need filters when running the content
+                # validator, which is the only time a model will have errors.
+                raise NameError(
```

Comment:
I usually see `NameError` reserved for [code-related issues](https://docs.python.org/3/library/exceptions.html#NameError), not runtime stuff like this one. I would prefer raising a `TypeError` here, since that's the problem (`filters` can't be of type `None`).